src/server/error/error.go

```diff
@@ -0,0 +1,53 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package error
+
+import (
+	"errors"
+	ierror "github.com/goharbor/harbor/src/internal/error"
+	"net/http"
+)
+
+var (
+	codeMap = map[string]int{
```

Comment:
don't like a code map defined, no better ways?

Reply:
As our new error code has a fixed relationship with the HTTP code. Maybe it's better to introduce the error decorator interface back to handle such a relationship.

@wy65701436